recipes/libcf/build.sh

```diff
@@ -0,0 +1,15 @@
+export CFLAGS="-Wall -g -m64 -pipe -O2  -fPIC"
+export CXXLAGS="${CFLAGS}"
+export CPPFLAGS="-I${PREFIX}/include"
+export LDFLAGS="-L${PREFIX}/lib"
+export LFLAGS="-fPIC"
+
+CONDA_LST=`conda list`
+if [[ ${CONDA_LST}'y' == *'openmpi'* ]]; then
```

Comment:
@msarahan @minrk I would love to hear back from you on this. It is indeed a thorny point with conda and our system. If you want to take adavantage of mpi then some software needs to be *compiled* in  a special fashion. Easiest example is hdf5. If I want to take advantage on hdf5 mpi's capabilities then I need hdf5 to be compiled against mpi. And flavors probably matter to. The way we solve this in our system is that my meta.yaml are actually pre-processed by a prep_for_build.py script (see: https://github.com/UV-CDAT/conda-recipes/blob/master/prep_for_build.py ) that make sure the dependencies are then set properly. It does *smell* a lot like conda's `feature` thing but it's not really. The only way to support different implementation is to add the "feature" name to the package for example using the same meta.yaml.in I generate either a meta.yaml that will be producing "mypackage" or "mypackage-openmpi" I implemented a {} system similar to conda's [] to support this. It is not ideal.
In the case of conda-forge I'm ok with offering two different recipes one for the "Standard" build one for the "package-openmpi" or "package-mpch" etc... But having a conda-proper way of doing things would be much better.